src/utils/settings/settingsFileTransaction.ts

```
@@ -0,0 +1,329 @@
+  const release = acquireSettingsLock(targetPath)
+  let operationFailed = false
+  try {
+    return operation(targetPath)
+  } catch (error) {
+    operationFailed = true
+    throw error
+  } finally {
+    try {
+      release()
+    } catch (releaseError) {
+      if (!operationFailed) throw releaseError
```

Biome (2.5.6)

[error] Unsafe usage of 'throw'.

(lint/correctness/noUnsafeFinally)